src/kwlib.sh

```diff
@@ -322,3 +322,19 @@ function command_exists()
   fi
   return 22 # EINVAL
 }
+
+# This function exits with a custom error message
+#
+# @err The error code to be used on exit
+# @msg The custom message to be displayed
+function exit_msg()
+{
+  local err="$1"
+  local msg="$2"
+
+  err=${err:-1}
```

Comment:
why are you taking 1 as the default error code?

Reply:
1 means 'Operation not permitted", which does not make sense here. Maybe 22 is a better exit value.